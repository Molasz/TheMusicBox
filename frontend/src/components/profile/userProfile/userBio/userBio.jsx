import React from 'react';
import { connect } from 'react-redux';

import './userBio.scss';

import { editBio, editName } from '../../../../redux/actions/authActions';

function UserBio({ bio, name, editInfo, dispatch }) {
  return (
    <section className='bio'>
      <div className='bio__top'>
        {!(
          Object.keys(editInfo).length === 0 && editInfo.constructor === Object
        ) ? (
          <input
            className='title__name'
            value={editInfo.name}
            onChange={(event) => dispatch(editName(event.target.value))}
          />
        ) : (
          <strong className='title__name'>{name}</strong>
        )}
      </div>
      <div className='bio__bottom'>
        {!(
          Object.keys(editInfo).length === 0 && editInfo.constructor === Object
        ) ? (
          <textarea
            className='bottom__text'
            value={editInfo.bio}
            onChange={(event) => dispatch(editBio(event.target.value))}
          />
        ) : (
          <p className='bottom__text'>{bio}</p>
        )}
      </div>
    </section>
  );
}

function mapDispatchToProps(state) {
  return { editInfo: state.authReducer.editInfo };
}

export default connect(mapDispatchToProps)(UserBio);
export { UserBio };
