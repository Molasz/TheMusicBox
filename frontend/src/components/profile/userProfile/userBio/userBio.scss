$color-detail: #f55110;
$main-color: #1a1a1a;
.user-bio {
  display: flex;
  flex-direction: column;

  width: 200px;
  margin-right: 30px;
  .user-bio__top {
    display: flex;
    justify-content: center;

    border-bottom: 3px solid $color-detail;
    padding: 10px 0;
    .top__input {
      background-color: $main-color;
      color: white;
      width: 100%;
      font-family: Arial;
      font-weight: bold;

      border: 2px solid $color-detail;
      border-radius: 5%;

      text-align: center;
      text-transform: capitalize;
      font-size: 25px;

      margin: 0;
      padding: 0;
      &:focus {
        outline: none;
      }
    }

    .top__name {
      text-transform: capitalize;
      font-size: 25px;
      text-align: center;
    }
  }

  .user-bio__middle {
    display: flex;
    justify-content: center;

    margin: 5px 10px;
    .middle__text-area {
      background-color: $main-color;
      color: white;
      font-family: Arial;

      border: 2px solid $color-detail;
      border-radius: 5%;

      font-size: 15px;

      padding: 15px 5px;
      min-width: 170px;
      height: 150px;
      &:focus {
        outline: none;
      }
    }

    .middle__bio {
      font-size: 15px;

      &::first-letter {
        text-transform: capitalize;
      }
    }
  }

  .user-bio__bottom {
    display: flex;
    align-items: center;
    flex-direction: column;

    padding: 0 10px;

    border-top: 3px solid $color-detail;
    border-bottom: 3px solid $color-detail;

    .bottom__text {
      text-align: center;
      font-size: 20px;
      margin: 10px;
    }

    .bottom__icon {
      font-size: 80px;
      color: $color-detail;
    }
  }
}
