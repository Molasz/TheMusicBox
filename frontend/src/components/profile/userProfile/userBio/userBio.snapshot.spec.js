import React from 'react';
import renderer from 'react-test-renderer';

import { UserBio } from './userBio';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('Bio snapshot', () => {
  const bio = '...';
  const name = 'name';

  it('Should match whitout edit info', () => {
    const editInfo = {};
    const tree = renderer.create(
      <UserBio bio={bio} name={name} editInfo={editInfo} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should match with edit info', () => {
    const editInfo = { user: 'user', bio: 'bio' };
    const tree = renderer.create(
      <UserBio bio={bio} name={name} editInfo={editInfo} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  const dispatch = jest.fn();
  const editInfo = { user: 'user', bio: 'bio' };
  const document = shallow(
    <UserBio dispatch={dispatch} bio={bio} name={name} editInfo={editInfo} />
  );

  it('Should call dispatch when change input', () => {
    const button = document.find('.top__input');
    button.simulate('change', { target: { value: 1 } });

    expect(dispatch.call).truthy;
  });

  it('Should call dispatch when change textArea', () => {
    const button = document.find('.bottom__text-area');
    button.simulate('change', { target: { value: 1 } });

    expect(dispatch.call).truthy;
  });
});
