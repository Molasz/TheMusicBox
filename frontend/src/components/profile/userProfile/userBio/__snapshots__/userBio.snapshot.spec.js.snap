// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio snapshot Should match whitout edit info 1`] = `
<section
  className="user-bio"
>
  <div
    className="user-bio__top"
  >
    <strong
      className="top__name"
    >
      name
    </strong>
  </div>
  <div
    className="user-bio__bottom"
  >
    <p
      className="bottom__bio"
    >
      ...
    </p>
  </div>
</section>
`;

exports[`Bio snapshot Should match with edit info 1`] = `
<section
  className="user-bio"
>
  <div
    className="user-bio__top"
  >
    <input
      className="top__input"
      onChange={[Function]}
      value="user"
    />
  </div>
  <div
    className="user-bio__bottom"
  >
    <textarea
      className="bottom__text-area"
      onChange={[Function]}
      value="bio"
    />
  </div>
</section>
`;
